src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseParser.cs

```diff
@@ -286,47 +296,69 @@ private static string GetReasonPhrase(HttpStatusCode statusCode, char[] buffer,
                 new string(buffer, 0, bufferLength);
         }
 
-        private static void ParseResponseHeaders(
+        private class HttpResponseTrailers : HttpHeaders
+        {
+        }
+
+        public static void ParseResponseHeaders(
             SafeWinHttpHandle requestHandle,
+            uint infoLevel,
             HttpResponseMessage response,
             char[] buffer,
-            bool stripEncodingHeaders)
+            bool stripEncodingHeaders,
+            bool isTrailers)
         {
             HttpResponseHeaders responseHeaders = response.Headers;
             HttpContentHeaders contentHeaders = response.Content.Headers;
+#if NETSTANDARD2_1 || NET6_0
+            HttpResponseHeaders responseTrailers = response.TrailingHeaders;
```

Comment:
Maybe we can set `response.RequestMessage.Properties["__ResponseTrailers"]` even in this case, so there are no breaking changes when a user upgrades?

Reply:
I was thinking about the same ... I assume the perf overhead is minimal

Reply:
I disagree here -- I don't think we should make all .NET 6 users pay for a feature that will only be used in gRPC. This should be a gRPC-only overhead that is tested for at runtime.

`Properties` is lazily allocated currently and generally is not used by anyone other than Blazor.